### PR TITLE
Fix the issue of suffixes 'f' and 'd' appearing two times in compilation errors

### LINKED ISSUE #35152: The letters 'f' and 'd' appear two time in compilation errors
**Description:**

The suffix 'f' and 'd' appears two time in compilation errors for the cases which involve with floating point and decimal literals.

**Steps to reproduce:**
```ballerina
public function main() {
    1f _ = 5; // incompatible types: expected '1ff', found 'int'

    1d _ = 5; // incompatible types: expected '1dd', found 'int'
}
```

**Affected Versions:**
2201.0.0

### PR DESCRIPTION
**Pull request by SandaruJayawardana**
Fixes #35152 

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeBuilder.java [hunk 1]
```diff
@@ -5350,7 +5350,7 @@ private BLangLiteral createSimpleLiteral(Node literal, SyntaxKind sign, boolean
                     value = textValue.replaceAll("[fd+]", "");
                     originalValue = textValue.replace("+", "");
                 } else {
-                    value = textValue;
+                    value = textValue.replaceAll("[fd+]", "");
```
Comment (chiranSachintha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeBuilder.java:5353]: `1f f1 = 2;` 
In the above example, `1f` is a finite type. Right?
In L1353 we create `BLangFiniteTypeNode` and use `createSimpleLiteral(Node literal)` function to create literal value. Here we set the `isFiniteType` flag as false. Isn't it the issue for this?

Comment (SandaruJayawardana) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeBuilder.java:5353]: My bad. Will update the fix. Thanks.

File: tests/jballerina-unit-test/src/test/resources/test-src/types/finite/finite_type_negative.bal [hunk 2]
```diff
@@ -118,3 +118,19 @@ function testFiniteTypeWithNumericValues() {
...
+    0x1p-1 f5 = 1.2f;
+    0x.12p12 f6 = 1.2d;
+    1.2e12 f7 = 2.1d;
+
+    2d d1 = 2;
+    1.2d d2 = 2.1;
+    1.21d d3 = 1.2f;
+    0.1219e-1 d4 = 1.21d;
+    12.1d d5 = 0x.21;
+}
```
Comment (rdulmina) [inline tests/jballerina-unit-test/src/test/resources/test-src/types/finite/finite_type_negative.bal:136]: Let's add a few more tests in different contexts.

eg:
```bal
public function main() {
    1f _ = <1f>5; // type cast expression context
    name(3); // argument context
}

function name(1d a) {
}

**Comment by rdulmina** [comment]
`TypeGuardTest` is failing let's fix it

File: tests/jballerina-unit-test/src/test/resources/test-src/types/finite/finite_type_negative.bal [hunk 3]
```diff
@@ -118,3 +118,49 @@ function testFiniteTypeWithNumericValues() {
...
+
+2d d1 = 4;
+1.2d d2 = 2.1;
+1.21d d3 = 1.2f;
+0.1219e-1 d4 = 1.21d;
+12.1d d5 = 0x.21;
+
+public function main() {
+
+}
```
Comment (gimantha) [inline tests/jballerina-unit-test/src/test/resources/test-src/types/finite/finite_type_negative.bal:166]:
```suggestion
}

```

Comment (rdulmina) [inline tests/jballerina-unit-test/src/test/resources/test-src/types/finite/finite_type_negative.bal:166]: Why do we need a main here?

File: tests/jballerina-unit-test/src/test/resources/test-src/types/finite/finite_type_negative.bal [hunk 4]
```diff
@@ -118,3 +118,49 @@ function testFiniteTypeWithNumericValues() {
     Foo4 f5 = 3;
     "chiran" x = 5;
 }
+
+function testFiniteTypeOfNumericValues() {
+    1f f1 = 2;
```
Comment (rdulmina) [inline tests/jballerina-unit-test/src/test/resources/test-src/types/finite/finite_type_negative.bal:123]: Let's use `_` for unused variables

File: tests/jballerina-unit-test/src/test/resources/test-src/types/finite/finite_type_negative.bal [hunk 5]
```diff
@@ -118,3 +118,49 @@ function testFiniteTypeWithNumericValues() {
...
+1.2e12 f4 = 2.1;
+0x1p-1 f5 = 1.2f;
+0x.12p12 f6 = 1.2d;
+1.2e12 f7 = 2.1d;
+
+2d d1 = 4;
+1.2d d2 = 2.1;
+1.21d d3 = 1.2f;
+0.1219e-1 d4 = 1.21d;
+12.1d d5 = 0x.21;
```
Comment (rdulmina) [inline tests/jballerina-unit-test/src/test/resources/test-src/types/finite/finite_type_negative.bal:162]: What I meant was to add tests where the type desc is in different contexts.  eg: typecast param ctx, function parameter ctx.

File: tests/jballerina-unit-test/src/test/resources/test-src/types/finite/finite_type_negative.bal (hunk 3 above)
Comment (SandaruJayawardana) [reply tests/jballerina-unit-test/src/test/resources/test-src/types/finite/finite_type_negative.bal:166]: my bad. Forgot to remove it.

File: tests/jballerina-unit-test/src/test/resources/test-src/types/finite/finite_type_negative.bal (hunk 5 above)
Comment (SandaruJayawardana) [reply tests/jballerina-unit-test/src/test/resources/test-src/types/finite/finite_type_negative.bal:162]: Will add them too. Thanks.

**Review by chiranSachintha — APPROVED** [review]

**Review by rdulmina — APPROVED** [review]

**Pull request merged**